src/main/java/javaslang/collection/Vector.java

```diff
@@ -0,0 +1,1481 @@
+/*     / \____  _    ______   _____ / \____   ____  _____
+ *    /  \__  \/ \  / \__  \ /  __//  \__  \ /    \/ __  \   Javaslang
+ *  _/  // _\  \  \/  / _\  \\_  \/  // _\  \  /\  \__/  /   Copyright 2014-2015 Daniel Dietrich
+ * /___/ \_____/\____/\_____/____/\___\_____/_/  \_/____/    Licensed under the Apache License, Version 2.0
+ */
+package javaslang.collection;
+
+import javaslang.Kind;
+import javaslang.Lazy;
+import javaslang.Tuple;
+import javaslang.Tuple2;
+import javaslang.control.None;
+import javaslang.control.Option;
+import javaslang.control.Some;
+
+import java.io.*;
+import java.util.*;
+import java.util.function.*;
+import java.util.stream.Collector;
+
+/**
+ * TODO javadoc
+ * @param <T>
+ */
+public interface Vector<T> extends IndexedSeq<T> {
```

Comment:
Please remove all public API related to Trie. We do not really need it. The new `Value` I committed yesterday has all conversion methods between 'regular' collection one could need. I see HAMT, RedBlackTree, ... as backing collections, not as core collections. They do not extends Traversable et al. If in doubt, we have implemented the Iterable interface, which is sufficient to create any Traversable collection via `ofAll`.